Fixes Portal Height

westonCoder opened the pull request:
- I am not really sure if this is the exact value you were going for with this, but 1.6 seems to match the collision to the top and bottom of the portal cutout.
- larger items like boxes seem to pass through much easier now.
- have not seen any negative side effect from this.

Fixes #294
also seems to fix the following now:
Fixes #260

lambertjamesd commented:
I think we should half the width instead then change
`gPortalOutline` so it doesn't multiply the X value for each entry by 0.5

westonCoder commented:
@lambertjamesd I believe this is now fixed. 
I changed the variable names to include _RADIUS 
and also changed some surrounding code . 

there were some lingering 0.8xxxx and 0.3xxxxx which I removed that I believe were from a previous iteration. although the negative values I replaced with -1.0 so that they were still being multiplied by a negative.

tested with a box again and it seems to pass through no problem.

lambertjamesd commented on src/scene/portal.c:
```diff
@@ -30,14 +30,14 @@ struct ColliderTypeData gPortalColliderType = {
+    {0.5f * SCENE_SCALE * PORTAL_COVER_WIDTH_RADIUS, 0.0f, 0},
+    {SCENE_SCALE * PORTAL_COVER_WIDTH_RADIUS, -1.0f * SCENE_SCALE * PORTAL_COVER_HEIGHT_RADIUS, 0},
+    {0.0f, -1.0f * SCENE_SCALE * PORTAL_COVER_HEIGHT_RADIUS, 0},
+    {-1.0f * SCENE_SCALE * PORTAL_COVER_WIDTH_RADIUS, -1.0f * SCENE_SCALE * PORTAL_COVER_HEIGHT_RADIUS, 0},
```
.353553f should become .707107f
And 0.5 to 1.0

westonCoder replied:
what should the original 0.707107f's become? just 1.0f?

lambertjamesd commented on src/scene/render_plan.c:
```diff
@@ -266,7 +266,7 @@ int renderPlanPortal(struct RenderPlan* renderPlan, struct Scene* scene, struct
     quatMultVector(&next->camera.transform.rotation, &gForward, &cameraForward);
 
-    next->camera.nearPlane = (-vector3Dot(&portalOffset, &cameraForward)) * SCENE_SCALE - SCENE_SCALE * PORTAL_COVER_HEIGHT * 0.5f;
+    next->camera.nearPlane = (-vector3Dot(&portalOffset, &cameraForward)) * SCENE_SCALE - SCENE_SCALE * PORTAL_COVER_HEIGHT_RADIUS * 0.5f;
```
Remove the 0.5f multiply here

lambertjamesd commented on src/scene/portal.c:
```diff
@@ -86,12 +86,12 @@ void portalCalculateBB(struct Transform* portalTransform, struct Box3D* bb) {
 
     struct Vector3 nextDir;
-    vector3Scale(&portalUp, &nextDir, PORTAL_COVER_HEIGHT * 0.5f);
+    vector3Scale(&portalUp, &nextDir, PORTAL_COVER_HEIGHT_RADIUS * 0.5f);
```
Remove the 0.5 multiply here

lambertjamesd commented on src/scene/portal.c:
```diff
@@ -86,12 +86,12 @@ void portalCalculateBB(struct Transform* portalTransform, struct Box3D* bb) {
     box3DExtendDirection(bb, &nextDir, bb);
 
-    vector3Scale(&portalRight, &nextDir, PORTAL_COVER_WIDTH * 0.5f);
+    vector3Scale(&portalRight, &nextDir, PORTAL_COVER_WIDTH_RADIUS * 0.5f);
```
Remove the 0.5 multiply here

westonCoder commented:
@lambertjamesd I think I made all requested changes properly. video attached. not seeing anymore artifacts on the sides of portals.

https://github.com/lambertjamesd/portal64/assets/71656782/3441dc87-2bb6-40ea-a2ae-1ab7b072895a
